src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+			case RatingsPromptStage.RATE:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.Rate");
+			case RatingsPromptStage.FEEDBACK:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.Feedback");
+			case RatingsPromptStage.END:
+				return Localization.getLocalizedString("WebClipper.Label.Ratings.Message.End");
+			default:
+			case RatingsPromptStage.NONE:
+				break;
+		}
+	}
+
+	/**
+	 * Get appropriate dialog panel buttons for the panel (with its internal states) provided
+	 */
+	public static getDialogButtons(panel: SuccessPanelClass): DialogButton[] {
+		let stage: RatingsPromptStage = panel.state.userSelectedRatingsPromptStage;
+		let clipperState: ClipperState = panel.props.clipperState;
+		let clientType: ClientType = clipperState.clientInfo.clipperType;
+
+		let buttons: DialogButton[] = [];
+
+		switch (stage) {
+			case RatingsPromptStage.INIT:
+				buttons.push({
+					id: Constants.Ids.ratingsButtonInitYes,
+					label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.Init.Positive"),
+					handler: () => {
+						RatingsHelper.setDoNotPromptStatus();
+
+						let rateUrl: string = RatingsHelper.getRateUrlIfExists(clientType);
+						if (!Utils.isNullOrUndefined(rateUrl) && rateUrl.length > 0) {
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.RATE
+							});
+						} else {
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.END
+							});
+						}
+					}
+				}, {
+						id: Constants.Ids.ratingsButtonInitNo,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.Init.Negative"),
+						handler: () => {
+							Clipper.Storage.getValue(Constants.StorageKeys.lastSeenVersion, (lastSeenVersion) => {
+								RatingsHelper.setLastBadRating(Date.now().toString(), lastSeenVersion).then((badRatingAlreadyOccurred) => {
+									if (badRatingAlreadyOccurred) {
+										// setting this to prevent additional ratings prompts after the second bad rating
+										RatingsHelper.setDoNotPromptStatus();
+									}
+								}, () => {
+									// values to set were invalid: err on the side of caution, always set do not prompt status
+									RatingsHelper.setDoNotPromptStatus();
+								}).then(() => {
+									let feedbackUrl: string = RatingsHelper.getFeedbackUrlIfExists(clipperState);
+									if (!Utils.isNullOrUndefined(feedbackUrl) && feedbackUrl.length > 0) {
+										panel.setState({
+											userSelectedRatingsPromptStage: RatingsPromptStage.FEEDBACK
+										});
+									} else {
+										panel.setState({
+											userSelectedRatingsPromptStage: RatingsPromptStage.END
+										});
+									}
+								});
+							});
+						}
+					});
+				break;
+			case RatingsPromptStage.RATE:
+				let rateUrl: string = RatingsHelper.getRateUrlIfExists(clientType);
+				if (!Utils.isNullOrUndefined(rateUrl) && rateUrl.length > 0) {
```

Comment:
oh wait, I just realized I misinterpreteted what I tested in the console - yeah you're right #Pending